ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/question_answering/sentence/DefaultSentenceSegmenter.java

```diff
@@ -0,0 +1,269 @@
+        }
+    }
+
+    /**
+     * Creates a new DefaultSentenceSegmenter with default configuration.
+     */
+    public DefaultSentenceSegmenter() {
+        this.defaultConfig = SentenceSegmentationConfig.getDefault();
+    }
+
+    /**
+     * Creates a new DefaultSentenceSegmenter with custom configuration.
+     *
+     * @param config The configuration to use
+     */
+    public DefaultSentenceSegmenter(SentenceSegmentationConfig config) {
+        this.defaultConfig = config != null ? config : SentenceSegmentationConfig.getDefault();
+    }
+
+    @Override
+    public List<Sentence> segment(String text) {
+        return segment(text, defaultConfig);
+    }
+
+    @Override
+    public List<Sentence> segment(String text, SentenceSegmentationConfig config) {
+        return segmentSentences(text, config);
+    }
+
+    /**
+     * Segments text into sentences using the specified configuration.
+     *
+     * @param text The text to segment
+     * @param config The configuration to use
+     * @return List of Sentence objects
+     * @throws IllegalArgumentException if the input text is null or empty
+     */
+    private List<Sentence> segmentSentences(String text, SentenceSegmentationConfig config) {
+        config.validate();
+        List<Sentence> sentences = new ArrayList<>();
+        String[] rawSentences = splitIntoSentences(text, config);
+
+        log.debug("Split into {} raw sentences", rawSentences.length);
+
+        if (rawSentences.length == 0) {
+            throw new IllegalArgumentException("Failed to split text into sentences");
+        }
+
+        int currentPosition = 0;
+        int currentIndex = 0;
+
+        for (String rawSentence : rawSentences) {
+            String trimmedSentence = config.isPreserveWhitespace() ? rawSentence : rawSentence.trim();
+
+            if (trimmedSentence.length() >= config.getMinSentenceLength()) {
+                int startIndex = text.indexOf(rawSentence, currentIndex);
+                if (startIndex == -1) {
+                    // Fallback to a more robust search if exact match fails
+                    startIndex = findApproximatePosition(text, rawSentence, currentIndex);
+                }
+                int endIndex = startIndex + rawSentence.length();
+
+                Sentence.SentenceType type = determineSentenceType(trimmedSentence);
+
+                sentences
+                    .add(
+                        Sentence
+                            .builder()
+                            .text(trimmedSentence)
+                            .startIndex(startIndex)
+                            .endIndex(endIndex)
+                            .position(currentPosition)
+                            .type(type)
+                            .build()
+                    );
+
+                currentPosition++;
+                currentIndex = endIndex;
+            } else {
+                log
+                    .debug(
```

Comment:
when the sentence is too short, it quietly skip tokenize the sentence? is the the same behaviour for other text embedding models?

Reply:
I have updated logic without this limitation as of now.